3.2rc1 ASDF test_backwards_compat failed

```
============================= test session starts =============================
platform win32 -- Python 3.7.1, pytest-4.1.0, py-1.7.0, pluggy-0.8.0

Running tests with Astropy version 3.2rc1.
Running tests in Miniconda3\envs\py37\lib\site-packages\astropy.

Date: 2019-05-09T14:24:44

Platform: Windows-10-10.0.17763-SP0

Executable: C:\...\Miniconda3\envs\py37\python.exe

Full Python Version: 
3.7.1 (default, Oct 28 2018, 08:39:03) [MSC v.1912 64 bit (AMD64)]

encodings: sys: utf-8, locale: cp1252, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.16.2
Scipy: 1.1.0
Matplotlib: 3.0.3
h5py: 2.8.0
Pandas: 0.23.4
Asdf: 2.3.3
Cython: 0.29
Scikit-image: not available
astropy_helpers: 3.2rc1
Using Astropy options: remote_data: any.

Matplotlib: 3.0.3
Freetype: 2.9.1
rootdir: C:\..., inifile:
plugins: xdist-1.24.0, remotedata-0.3.1, openfiles-0.3.0, mpl-0.10, forked-0.2, doctestplus-0.1.3, arraydiff-0.2
collected 12639 items
```

```
____________________________ test_backwards_compat ____________________________

    def test_backwards_compat():
        """
        Make sure that we can continue to read FITS HDUs that use the schema from
        the ASDF Standard.
    
        This test uses the examples in the fits schema from the ASDF Standard,
        since these make no reference to Astropy's own fits definition.
        """
    
        def check(asdffile):
            assert isinstance(asdffile['example'], fits.HDUList)
    
>       run_schema_example_test('stsci.edu', 'asdf', 'fits/fits', '1.0.0', check)

..\Miniconda3\envs\py37\lib\site-packages\astropy\io\misc\asdf\tags\fits\tests\test_fits.py:55: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

organization = 'stsci.edu', standard = 'asdf', name = 'fits/fits'
version = '1.0.0'
check_func = <function test_backwards_compat.<locals>.check at 0x000002316C136730>

    def run_schema_example_test(organization, standard, name, version, check_func=None):
    
        import asdf
        from asdf.tests import helpers
        from asdf.types import format_tag
        from asdf.resolver import default_resolver
    
        tag = format_tag(organization, standard, version, name)
        schema_path = urllib.parse.urlparse(default_resolver(tag)).path
    
>       with open(schema_path, 'rb') as ff:
E       OSError: [Errno 22] Invalid argument: '/C:/Users/PL/Miniconda3/envs/py37/lib/site-packages/asdf/schemas/stsci.edu/asdf/fits/fits-1.0.0.yaml'

..\Miniconda3\envs\py37\lib\site-packages\astropy\io\misc\asdf\tags\tests\helpers.py:21: OSError
____________________________ test_backwards_compat ____________________________

    def test_backwards_compat():
        """
        Make sure that we can continue to read tables that use the schema from
        the ASDF Standard.
    
        This test uses the examples in the table schema from the ASDF Standard,
        since these make no reference to Astropy's own table definition.
        """
    
        def check(asdffile):
            assert isinstance(asdffile['example'], table.Table)
    
>       run_schema_example_test('stsci.edu', 'asdf', 'core/table', '1.0.0', check)

..\Miniconda3\envs\py37\lib\site-packages\astropy\io\misc\asdf\tags\table\tests\test_table.py:244: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

organization = 'stsci.edu', standard = 'asdf', name = 'core/table'
version = '1.0.0'
check_func = <function test_backwards_compat.<locals>.check at 0x000002316C157D90>

    def run_schema_example_test(organization, standard, name, version, check_func=None):
    
        import asdf
        from asdf.tests import helpers
        from asdf.types import format_tag
        from asdf.resolver import default_resolver
    
        tag = format_tag(organization, standard, version, name)
        schema_path = urllib.parse.urlparse(default_resolver(tag)).path
    
>       with open(schema_path, 'rb') as ff:
E       OSError: [Errno 22] Invalid argument: '/C:/Users/PL/Miniconda3/envs/py37/lib/site-packages/asdf/schemas/stsci.edu/asdf/core/table-1.0.0.yaml'

..\Miniconda3\envs\py37\lib\site-packages\astropy\io\misc\asdf\tags\tests\helpers.py:21: OSError
```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -11,7 +11,7 @@ environment:
                           # of 32 bit and 64 bit builds are needed, move this
                           # to the matrix section.
         CONDA_DEPENDENCIES: "Cython scipy h5py beautifulsoup4 html5lib jinja2 pyyaml matplotlib scikit-image pytz"
-        PIP_DEPENDENCIES: "objgraph"
+        PIP_DEPENDENCIES: "objgraph asdf"
 
     matrix:
         - PYTHON_VERSION: "3.7"

--- a/astropy/io/misc/asdf/tags/tests/helpers.py
+++ b/astropy/io/misc/asdf/tags/tests/helpers.py
@@ -2,6 +2,7 @@
 # -*- coding: utf-8 -*-
 import os
 import urllib.parse
+import urllib.request
 
 import yaml
 
@@ -13,15 +14,15 @@ def run_schema_example_test(organization, standard, name, version, check_func=No
     import asdf
     from asdf.tests import helpers
     from asdf.types import format_tag
-    from asdf.resolver import default_resolver
+    from asdf.resolver import default_tag_to_url_mapping
+    from asdf.schema import load_schema
 
     tag = format_tag(organization, standard, version, name)
-    schema_path = urllib.parse.urlparse(default_resolver(tag)).path
-
-    with open(schema_path, 'rb') as ff:
-        schema = yaml.load(ff)
+    uri = asdf.resolver.default_tag_to_url_mapping(tag)
+    r = asdf.AsdfFile().resolver
 
     examples = []
+    schema = load_schema(uri, resolver=r)
     for node in asdf.treeutil.iter_tree(schema):
         if (isinstance(node, dict) and
             'examples' in node and
@@ -31,7 +32,7 @@ def run_schema_example_test(organization, standard, name, version, check_func=No
 
     for example in examples:
         buff = helpers.yaml_to_asdf('example: ' + example.strip())
-        ff = asdf.AsdfFile(uri=schema_path)
+        ff = asdf.AsdfFile(uri=uri)
         # Add some dummy blocks so that the ndarray examples work
         for i in range(3):
             b = asdf.block.Block(np.zeros((1024*1024*8), dtype=np.uint8))

--- a/astropy/io/misc/asdf/tags/tests/helpers.py
+++ b/astropy/io/misc/asdf/tags/tests/helpers.py
@@ -1,11 +1,5 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 # -*- coding: utf-8 -*-
-import os
-import urllib.parse
-import urllib.request
-
-import yaml
-
 import numpy as np
 
 
@@ -14,7 +8,6 @@ def run_schema_example_test(organization, standard, name, version, check_func=No
     import asdf
     from asdf.tests import helpers
     from asdf.types import format_tag
-    from asdf.resolver import default_tag_to_url_mapping
     from asdf.schema import load_schema
 
     tag = format_tag(organization, standard, version, name)

